src/Common/Auth/OpenIDConnect/Grant/CustomRefreshTokenGrant.php

```diff
@@ -0,0 +1,118 @@
+<?php
+
+/**
+ * CustomRefreshTokenGrant Handles the custom nature of some of our scope api requests to differentiate between the
+ * standard api and the regular api.  Since we don't have access to the old refresh token scopes when we are creating
+ * our scope repository, we initialize them here so we can use them in our scope repo.
+ * @package openemr
+ * @link      http://www.open-emr.org
+ * @author    Stephen Nielson <stephen@nielson.org>
+ * @copyright Copyright (c) 2020 Stephen Nielson <stephen@nielson.org>
+ * @license   https://github.com/openemr/openemr/blob/master/LICENSE GNU General Public License 3
+ */
+
+namespace OpenEMR\Common\Auth\OpenIDConnect\Grant;
+
+use DateInterval;
+use League\OAuth2\Server\Entities\ScopeEntityInterface;
+use League\OAuth2\Server\Exception\OAuthServerException;
+use League\OAuth2\Server\Grant\RefreshTokenGrant;
+use League\OAuth2\Server\Repositories\RefreshTokenRepositoryInterface;
+use League\OAuth2\Server\ResponseTypes\ResponseTypeInterface;
+use OpenEMR\Common\Auth\OpenIDConnect\Repositories\ScopeRepository;
+use OpenEMR\Common\Logging\SystemLogger;
+use Psr\Http\Message\ServerRequestInterface;
+
+class CustomRefreshTokenGrant extends RefreshTokenGrant
+{
+    public function __construct(RefreshTokenRepositoryInterface $refreshTokenRepository)
+    {
+        parent::__construct($refreshTokenRepository);
+    }
+
+    public function respondToAccessTokenRequest(
+        ServerRequestInterface $request,
+        ResponseTypeInterface $responseType,
+        DateInterval $accessTokenTTL
+    ) {
+        $client = $this->validateClient($request);
+        $oldRefreshToken = $this->validateOldRefreshToken($request, $client->getIdentifier());
+        SystemLogger::instance()->debug("CustomRefreshTokenGrant->respondToAccessTokenRequest() scope info", [
```

Comment:
not sure if i've said this yet, but this log debug mechanism is awesome

![2DV](https://user-images.githubusercontent.com/278968/103306305-1c2b1280-49c2-11eb-9878-1d827c5b7466.gif)